### PR TITLE
Enable Dynamic Rendering extension

### PR DESCRIPTION
**Pull request by Top61ly**
# **Enabe Dynamic Rendering extension in gore**

## **Description**

Dynamic Rendering makes our rendering code get rid of Renderpass and Subpass. 

---

### **Possible issues**

* All texture formats are hardcoded
* On TBDR hardware, we don't have any subpass rendering path

### CODE REVIEW

File: src/Graphics/Device.cpp [hunk 1]
```diff
@@ -260,16 +260,22 @@ Device::Device(PhysicalDevice physicalDevice) :
...
 
     enabledFeatures2.pNext = &bufferDeviceAddressFeatures;
 
     // Device extensions
     std::vector<vk::ExtensionProperties> deviceExtensionProperties = pd.enumerateDeviceExtensionProperties();
     m_EnabledDeviceExtensions.set();
     std::vector<const char*> enabledDeviceExtensions = BuildEnabledExtensions<VulkanDeviceExtensionBitset, VulkanDeviceExtension>(deviceExtensionProperties,
-                                                                                                                                  m_EnabledDeviceExtensions);
+                                                                                                                                  m_EnabledDeviceExtensions);                                                                                                                                
+    enabledDeviceExtensions.push_back(VK_KHR_DYNAMIC_RENDERING_EXTENSION_NAME);
```
Comment (shurunxuan) [inline src/Graphics/Device.cpp]: Add this to VulkanExtensions.inl?

File: src/Graphics/Instance.cpp [hunk 2]
```diff
@@ -85,16 +85,23 @@ Instance::Instance(App* app) :
...
     // Instance Creation
 #ifdef VK_API_VERSION_1_1
     m_ApiVersion = m_Context.enumerateInstanceVersion(); // TODO: potential crash
 #else
     m_ApiVersion = VK_API_VERSION_1_0;
 #endif
 
+    uint32_t minorVersion = VK_API_VERSION_MINOR(m_ApiVersion);
+    uint32_t majorVersion = VK_API_VERSION_MAJOR(m_ApiVersion);
+    if (majorVersion >=1 && minorVersion >= 3)
```
Comment (shurunxuan) [inline src/Graphics/Instance.cpp]: You can directly do m_ApiVersion >= VK_API_VERSION_1_3, and also this code seems to be redundant as the m_ApiVersion is already as high as possible

**Review by shurunxuan — COMMENTED** [review]
With this change, are we limiting the device support to those only supports dynamic rendering, with it being a mandatory path, instead of optional?

**Comment by Top61ly** [comment]
> With this change, are we limiting the device support to those only supports dynamic rendering, with it being a mandatory path, instead of optional?

Hope we can finish gore in Vulkan 1.4 version. So, I think it's fine to use Vulkan 1.3 core feature.
Inspired by the new Vulkan youtube video.
https://www.youtube.com/watch?v=0OqJtPnkfC8&t=2257s

image

File: src/Graphics/Instance.cpp (hunk 2 above)
Comment (Top61ly) [reply src/Graphics/Instance.cpp]: Deleted!

File: src/Graphics/Device.cpp (hunk 1 above)
Comment (Top61ly) [reply src/Graphics/Device.cpp]: Moved!

**Review by shurunxuan — APPROVED** [review]
LGTM

**Pull request merged**